### PR TITLE
Monitor site access settings

### PR DESCRIPTION
**Pull request by GabrielCousin**
## 📝 What is this PR

Reported [internally](https://phantombuster.atlassian.net/browse/GRF-95) - the extension can stop working if site access settings are not set accordingly. This includes:
- a notification to the user (when installing/updating the extension, when removing access)
- some Sentry reporting
- improving Sentry monitoring with [setting id or email if provided](https://phantombuster.atlassian.net/browse/GRF-96)

## 🏭 How does it work

See https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/permissions
- `getAll` -> returns an object with origins (the list of whitelisted URL) and permissions
- `onRemoved` -> handler when a setting is removed

### ✅ Positive testing (should succeed)

Disclaimer: notifications must be enabled at the system level!
-   Install the extension 👉 it should work
-   Disable the "site access" globally 👉 some notifications should pop up
-   Disable the "site access" from the menu 
image 👉 some notifications should pop up
-   Remove linkedin origin from the list 👉 some notifications should pop up
- Sentry errors are enriched https://phantombuster.sentry.io/issues/4504900826/?project=6698508&query=is%3Aunresolved&referrer=issue-stream&stream_index=0

### CODE REVIEW

File: src/shared/sentry.ts [hunk 1]
```diff
@@ -17,6 +19,7 @@ export function initSentry() {
 		// Sentry bases its price on the amount of events, so for now the decision is to ignore them.
 		ignoreErrors: [
 			"ResizeObserver loop limit exceeded",
+			"ResizeObserver loop completed with undelivered notifications",
```
Comment (val92130) [inline src/shared/sentry.ts:22]: 🎉

**Review by val92130 — APPROVED** [review]
nice, seems to work great!
LGTM

**Review by Hatscat — COMMENTED** [review]
I don't have the pop ups in Arc, when I disable the site accesses it's like if I don't have the extension, maybe I do not test correctly ^^"

And you forgot to bump the version in the manifest! 


image

**Comment by GabrielCousin** [comment]
@Hatscat thanks for the manifest reminder! 😮‍💨  That's correct, if notifications are not enabled there is no better way at the moment to let the user know but at least we can have some "monitoring" on how things are going

**Comment by Hatscat** [comment]
> if notifications are not enabled there is no better way at the moment to let the user know but at least we can have some "monitoring" on how things are going

Ok, understood 🙂

File: manifest.beta.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "PhantomBuster BETA",
-	"version_name": "1.1.8",
+	"version_name": "1.1.9",
 	"browser_specific_settings": {
 		"gecko": {
 			"update_url": "https://raw.githubusercontent.com/phantombuster/web-browser-extension/beta/beta-updates.json"
```
Comment (GabrielCousin) [inline manifest.beta.json:6]: @Hatscat Just to confirm, the beta link is expected to be updated in a further PR, right?

Comment (Hatscat) [reply manifest.beta.json:6]: in fact I feel it's better to update it before ( = in this PR)
(from the doc by Maxime:)
```
- Check that the link to the uploaded .xpi in the release matches exactly the one that was crafted in `beta-updates.json`
(if it does not, just update it in `beta-updates.json`, no need to rebuild the extension as this file is not a part of it)
```

**Review by Hatscat — DISMISSED** [review]
LGTM 💯

**Review by Hatscat — APPROVED** [review]

**Review by Hatscat — APPROVED** [review]

**Review by val92130 — DISMISSED** [review]
LGTM

**Review by DamienLaf — DISMISSED** [review]

**Review by Hatscat — APPROVED** [review]

**Review by val92130 — APPROVED** [review]
LGTM

**Review by DamienLaf — APPROVED** [review]

**Review by RiahiKarim — APPROVED** [review]

**Pull request merged**